modules/prediction/common/message_process.cc

```diff
@@ -143,11 +150,12 @@ void MessageProcess::ContainerProcess(
   // Insert perception_obstacles
   ptr_obstacles_container->Insert(perception_obstacles);
 
+  ObstaclesPrioritizer obstacles_prioritizer(container_manager);
   // Ignore some obstacles
-  ObstaclesPrioritizer::Instance()->AssignIgnoreLevel();
+  obstacles_prioritizer.AssignIgnoreLevel();
 
   // Scenario analysis
-  ScenarioManager::Instance()->Run();
+  ScenarioManager::Instance()->Run(container_manager.get());
```

Comment:
Just to double check, scenario manager can still run as singleton?

Reply:
I think we'd better avoid using singleton for ScenarioManager here because it has a private member variable inside. Also, to make the whole thing consistent.

Reply:
Ok.